.github/workflows/srpm.yaml

```diff
@@ -0,0 +1,19 @@
+name: generate SRPM
+on: [push, pull_request]
+jobs:
+  srpm:
+    runs-on: ubuntu-latest
+    container:
+      image: fedora:rawhide
+      options: --privileged
+    steps:
+      - run: dnf install -y packit git nodejs
+      - uses: actions/checkout@v4
+      - run: git config --global --add safe.directory /__w/aws-ofi-nccl/aws-ofi-nccl
+      - run: packit srpm
```

Comment:
I thought the reason you wanted to go with packit was it would allow a single spec/build config file to generate both debs and RPMs, but that does not seem to be the case looking at the PR. If so, what does pakit provide over rpmbuild?

Reply:
Packit is not a replacement for rpmbuild, nor is mock a replacement for rpmbuild. Both [dnf](https://github.com/rpm-software-management/dnf/blob/master/.packit.yaml) and [mock](https://github.com/rpm-software-management/mock/blob/main/.packit.yaml) both use packit for srpm generation.

The general flow in the fedora-like world is as follows:

1. use [tito](https://github.com/rpm-software-management/tito) to tag your releases.
2. use [packit](https://packit.dev/) to generate a srpm for that tag.
3a. use [mock](https://rpm-software-management.github.io/mock/) to read that srpm, install all of its build dependencies in a chroot, and then perform the rpm build in that isolated environment. Packit can drive this automatically for you with `packit build in-mock` and produces an installable rpm.
3b. use [koji](https://fedoraproject.org/wiki/Koji) to read that srpm and build it and make it available in official repositories. Packit can drive this automatically with `packit build in-koji`.
3c. use [copr](https://copr.fedorainfracloud.org/) to read that srpm and build it and make it available in an unofficial copr repositories. Packit can drive this automatically with `packit build in-copr`.

There exists a separate tool, `debbuild`, which can take an srpm, then build it and package it as a `.deb`. This is what we originally agreed on, and it works fine with the provided rpmspec in this PR, but the repacking made Brian nervous about breakage and so I implemented a PKGBUILD for use with makedeb instead. My preference is still to just use `debbuild` so that there is a single srpm and single specfile.

If this all sounds awfully complicated, I tend to agree, which is why I advocated for using cpack instead.

Reply:
again, we don't need any of that.  The goal here is just to have a .spec file generated from configure that can be used in the same way as Libfabric.  Same for debian build files.  We already have a mechanism internally for taking that spec file and making binary packages from it.

Reply:
You can't generate the specfile from `./configure` because the specfile is where `./configure` is called...

What you're suggesting requires one to install all of our build dependencies (a new enough version of libfabric, a cuda 12.x toolkit install, hwloc and automake and autoconf, etc.), just to run `./configure`, just to build a separate rpmspec which will install all those dependencies and call configure again?

Reply:
again, I want you to copy what we do with libfabric.  Where they have a spec file included in the tarball that is written by the configure used to create the makefiles for running `make dist`.  See https://github.com/ofiwg/libfabric/blob/main/libfabric.spec.in

Reply:
What you're suggesting is the exact wrong way to do this. It creates a specfile that is totally unusable by any distribution that wants to consume it, and that's why fedora (and al2023 by extension) completely ignore it and why there's a totally independent and [separate specfile](https://src.fedoraproject.org/rpms/libfabric/blob/rawhide/f/libfabric.spec) checked into dist-git that does it correctly.

Reply:
And if that was the goal, that would be fine. But that's not the goal here.